controller_manager/src/controller_manager.cpp

```diff
@@ -2322,6 +2322,12 @@ std::vector<std::string> ControllerManager::get_controller_names()
 
 void ControllerManager::read(const rclcpp::Time & time, const rclcpp::Duration & period)
 {
+  if (
+    periodicity_stats_.GetCount() >=
+    params_->diagnostics.threshold.controller_manager.periodicity.window_size)
+  {
+    periodicity_stats_.Reset();
+  }
```

Comment:
Same here, this logic doesn't work for the default option
See the below comments for suggestions